### PR TITLE
Don't allow claims if less than 14 days have passed

### LINKED ISSUE #72: [SUGGESTION] Add warning if still not claimable (by date)
**Is your feature request related to a problem? Please describe.**
Right now we try to claim and unless a server problem we try it anyway without checking the last claim date.

**Describe the solution you'd like**
Check last claim date > 14 days before attempting a claim. And throw a warning if this case happens.

### PR DESCRIPTION
**Pull request by kara-todd**
closes #72

### CODE REVIEW

**Comment by kara-todd** [comment]
@FerranMarin I added onto the existing function to prevent a second network call from being made. I also had a lot of trouble with `timedelta` comparisons... I kept ending with `(now - last_claim_date).days` being 1 day off... I just did the comparison in seconds to avoid timezone shenanigans.  

Please let me know if there are any changes needed.

File: axie-scholar-utilities/source/axie/utils.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 USER_AGENT = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1944.0 Safari/537.36" # noqa
 TIMEOUT_MINS = 5
+AXIE_CLAIM_COOLDOWN_SECONDS = 60 * 60 * 24 * 14 # Timezone agnostic comparisons
```
Comment (FerranMarin) [inline axie-scholar-utilities/source/axie/utils.py:16]: 60s * 60min * 24h * 14days right? maybe add somehting like (14 days in seconds) ? So it's even easier to understand?

**Comment by FerranMarin** [comment]
Couple picky comments, but nice PR overall :D

**Comment by FerranMarin** [comment]
On another note, you might want to re-thing the seconds approach if you change fromtimestamp for utcfromtimestamp that way it should be tz agnostic and work fine...
Also, I believe we must consider the option where the lastclaim has updated to today but the claim has not actually happened... we might need to contemplate that and check with the diff of slp blance reported by graphql and the ronin chain check_balance for that account...
Yes, with SM and its chain stuff is never that straight forward.

**Comment by kara-todd** [comment]
> Also, I believe we must consider the option where the last claim has updated to today but the claim has not actually happened... we might need to contemplate that and check with the diff of slp blance reported by graphql and the ronin chain check_balance for that account...

Would this case not be covered by these lines?

```
   in_game_total = int(response.json()['total'])
   wallet_total = check_balance(self.account)
   if in_game_total > wallet_total:
   return in_game_total - wallet_total
```

If the claim is different on chain, the `in_game_total - wallet_total` should return 0 or negative no?

**Comment by FerranMarin** [comment]
True...
I am thinking if it would be worth making this extra check skippable, as I belive there is still a case where we would have SLP to claim, but the last claimed date would be of today (so lambda less than 14 days), but it would have failed...
It is quite rare, but I guess I'd like to add the option --force or something on those lines to skip this datetime check and just go for it (only outputing a log msg saying something "this acc has probably been claimed already unless operation failed on the chain today").

Does this sound reasonable to you too?

**Comment by kara-todd** [comment]
@FerranMarin 

Hmm. So let me restate this to make sure I understand.

We need to a flag `--force` or similar to allow by-passing the date check. This is done in the event that the previous claim transaction has failed. (Something we can't determine) Is that correct? Alternatively, would we be able to look that up on chain somehow? Do a filter query for the last SLP claim within the given date-range and check the transaction status?

Maybe that's too much work XD

I think the flag option seems reasonable. Are you already reading in flags somehow?

**Comment by FerranMarin** [comment]
So... flags are something you can find in the files _cli there is one for trezor and one for non-trezor. Both on the source folder.
And yes... we could query the chain to see if there has been a successful claim today for that account, but I believe we would need a data node accessible for such a thing, ie, not the ones we use atm.
So, I believe this --force in case of "major need" is the best next thing.

**Pull request merged**